flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/expressions/converter/OverConvertRule.java

```diff
@@ -0,0 +1,195 @@
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlPostfixOperator;
+import org.apache.calcite.sql.SqlWindow;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.OrdinalReturnTypeInference;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.table.planner.expressions.converter.ExpressionConverter.extractValue;
+import static org.apache.flink.table.runtime.types.LogicalTypeDataTypeConverter.fromDataTypeToLogicalType;
+
+/**
+ * A {@link CallExpressionConvertRule} that converts {@link BuiltInFunctionDefinitions#OVER}.
+ */
+public class OverConvertRule implements CallExpressionConvertRule {
+
+	@Override
+	public Optional<RexNode> convert(
+		CallExpression call, ConvertContext context) {
+		List<Expression> children = call.getChildren();
+		if (call.getFunctionDefinition() == BuiltInFunctionDefinitions.OVER) {
+			FlinkTypeFactory typeFactory = context.getTypeFactory();
+			Expression agg = children.get(0);
+			SqlAggFunction aggFunc = agg.accept(new SqlAggFunctionVisitor(typeFactory));
+			RelDataType aggResultType = typeFactory.createFieldTypeFromLogicalType(
+				fromDataTypeToLogicalType(((ResolvedExpression) agg).getOutputDataType()));
+
+			// assemble exprs by agg children
+			List<RexNode> aggExprs = agg.getChildren().stream().map(context::toRexNode)
+				.collect(Collectors.toList());
+
+			// assemble order by key
+			Expression orderKeyExpr = children.get(1);
+			Set<SqlKind> kinds = new HashSet<>();
+			RexNode collationRexNode = createCollation(
+				context.toRexNode(orderKeyExpr), RelFieldCollation.Direction.ASCENDING, null, kinds);
+			ImmutableList<RexFieldCollation> orderKey = ImmutableList
+				.of(new RexFieldCollation(collationRexNode, kinds));
+
+			// assemble partition by keys
+			List<RexNode> partitionKeys = children.subList(4, children.size()).stream().map(context::toRexNode)
+				.collect(Collectors.toList());
+			// assemble bounds
+			Expression preceding = children.get(2);
+			boolean isPhysical = LogicalTypeChecks.hasRoot(
+				fromDataTypeToLogicalType(((ResolvedExpression) preceding).getOutputDataType()),
+				LogicalTypeRoot.BIGINT);
+			Expression following = children.get(3);
+			RexWindowBound lowerBound = createBound(context, preceding, SqlKind.PRECEDING);
+			RexWindowBound upperBound = createBound(context, following, SqlKind.FOLLOWING);
+
+			// build RexOver
+			return Optional.of(context.getRelBuilder().getRexBuilder().makeOver(
+				aggResultType,
+				aggFunc,
+				aggExprs,
+				partitionKeys,
+				orderKey,
+				lowerBound,
+				upperBound,
+				isPhysical,
+				true,
+				false,
+				false));
+		}
+		return Optional.empty();
+	}
+
+	private RexNode createCollation(RexNode node, RelFieldCollation.Direction direction,
```

Comment:
Could we break the parameters according to our code style guidelines: https://flink.apache.org/contributing/code-style-and-quality-formatting.html#breaking-the-lines-of-too-long-statements ?

Reply:
You are right, got it.